gnu-apl-network.el

```diff
@@ -43,8 +58,7 @@ connect mode in use."
 
 (defun gnu-apl--connect (connect-mode addr)
   (with-current-buffer (gnu-apl--get-interactive-session)
-    (when (and (boundp 'gnu-apl--connection)
-               (process-live-p gnu-apl--connection))
+    (when (process-live-p gnu-apl--connection)
```

Comment:
What is the reason for this change? The binding of `gnu-apl--connection` in the buffer only was intentional. Granted, the behaviour should be the same, but it's nice not to pollute the namespace with symbols that can never be used outside an apl buffer.

Reply:
Do you have details?

Reply:
It turns out that declaring `gnu-apl--connection` is enough to solve the issue, in which case the boundp isn't necessary. Without declaring the variable, you get a number of byte compilation issues along the lines of "Warning: reference to free variable ...".

Reply:
Sure, but then we end up with the symbol being leaked outside the buffers where they belong. I blame the compiler here.

That said, as long as it works, it doesn't really matter, so just go ahead.